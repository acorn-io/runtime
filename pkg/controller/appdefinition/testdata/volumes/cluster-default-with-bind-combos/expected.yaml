kind: Deployment
apiVersion: apps/v1
metadata:
  name: container-name
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "container-name"
    "acorn.io/managed": "true"
spec:
  replicas: 1
  selector:
    matchLabels:
      "acorn.io/app-namespace": "app-namespace"
      "acorn.io/app-name": "app-name"
      "acorn.io/container-name": "container-name"
      "acorn.io/managed": "true"
  strategy:
    type: Recreate
  template:
    metadata:
      labels:
        "acorn.io/app-namespace": "app-namespace"
        "acorn.io/app-name": "app-name"
        "acorn.io/container-name": "container-name"
        "acorn.io/managed": "true"
      annotations:
        acorn.io/container-spec: '{"dirs":{"/var/temp":{"secret":{},"volume":"bar"},"/var/tmp":{"secret":{},"volume":"foo"}},"image":"image-name","probes":null}'
    spec:
      imagePullSecrets:
        - name: container-name-pull-1234567890ab
      terminationGracePeriodSeconds: 5
      hostname: container-name
      enableServiceLinks: false
      serviceAccountName: container-name
      volumes:
        - name: bar
          persistentVolumeClaim:
            claimName: bar
        - name: foo
          persistentVolumeClaim:
            claimName: foo
      containers:
        - name: container-name
          image: "image-name"
          volumeMounts:
            - mountPath: "/var/temp"
              name: bar
            - mountPath: "/var/tmp"
              name: foo
---
kind: PodDisruptionBudget
apiVersion: policy/v1
metadata:
  name: container-name
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "container-name"
    "acorn.io/managed": "true"
spec:
  selector:
    matchLabels:
      "acorn.io/app-namespace": "app-namespace"
      "acorn.io/app-name": "app-name"
      "acorn.io/container-name": "container-name"
      "acorn.io/managed": "true"
  maxUnavailable: 1
---
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: "foo"
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/managed": "true"
    "acorn.io/volume-name": "foo"
    acorn.io/volume-class: "test-custom-class"
spec:
  resources:
    requests:
      storage: 3Gi
  storageClassName: "custom-class"
  accessModes:
    - ReadWriteOnce
---
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: "bar"
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/managed": "true"
    "acorn.io/volume-name": "bar"
    acorn.io/volume-class: "test-custom-class"
spec:
  resources:
    requests:
      storage: 6Gi
  storageClassName: "custom-class"
  accessModes:
    - ReadWriteMany
---
kind: AppInstance
apiVersion: internal.acorn.io/v1
metadata:
  name: app-name
  namespace: app-namespace
  uid: 1234567890abcdef
spec:
  image: test
  volumes:
    - target: foo
      class: test-custom-class
      size: 3Gi
      accessModes: [ "readWriteOnce" ]
    - target: bar
      class: test-custom-class
status:
  namespace: app-created-namespace
  appImage:
    id: test
  appSpec:
    containers:
      container-name:
        image: "image-name"
        dirs:
          "/var/tmp":
            volume: foo
          "/var/temp":
            volume: bar
    volumes:
      foo:
        class: test-custom-class-request
        size: 6Gi
        accessModes: [ "readOnlyMany" ]
      bar:
        class: test-custom-class-request
        size: 6Gi
  conditions:
    - type: defined
      reason: Success
      status: "True"
      success: true
  defaults:
    volumes:
      foo:
        class: test-custom-class-default
        size: 2Gi
        accessModes: [ "readWriteMany" ]
      bar:
        class: test-custom-class-default
        accessModes: [ "readWriteMany" ]
