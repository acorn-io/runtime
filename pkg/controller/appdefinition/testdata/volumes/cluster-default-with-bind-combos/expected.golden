`apiVersion: v1
data:
  .dockerconfigjson: eyJhdXRocyI6eyJpbmRleC5kb2NrZXIuaW8iOnsiYXV0aCI6Ik9nPT0ifX19
kind: Secret
metadata:
  creationTimestamp: null
  labels:
    acorn.io/managed: "true"
    acorn.io/pull-secret: "true"
  name: container-name-pull-1234567890ab
  namespace: app-created-namespace
type: kubernetes.io/dockerconfigjson

---
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    acorn.io/config-hash: ""
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/app-public-name: app-name
    acorn.io/container-name: container-name
    acorn.io/managed: "true"
    acorn.io/project-name: app-namespace
  name: container-name
  namespace: app-created-namespace

---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    acorn.io/config-hash: ""
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/app-public-name: app-name
    acorn.io/container-name: container-name
    acorn.io/managed: "true"
    acorn.io/project-name: app-namespace
  name: container-name
  namespace: app-created-namespace
spec:
  replicas: 1
  selector:
    matchLabels:
      acorn.io/app-name: app-name
      acorn.io/app-namespace: app-namespace
      acorn.io/container-name: container-name
      acorn.io/managed: "true"
  strategy:
    type: Recreate
  template:
    metadata:
      annotations:
        acorn.io/container-spec: '{"dirs":{"/var/temp":{"secret":{},"volume":"bar"},"/var/tmp":{"secret":{},"volume":"foo"}},"image":"image-name","metrics":{},"probes":null}'
        karpenter.sh/do-not-evict: "true"
      creationTimestamp: null
      labels:
        acorn.io/app-name: app-name
        acorn.io/app-namespace: app-namespace
        acorn.io/app-public-name: app-name
        acorn.io/container-name: container-name
        acorn.io/managed: "true"
        acorn.io/project-name: app-namespace
    spec:
      containers:
      - image: image-name
        name: container-name
        resources: {}
        volumeMounts:
        - mountPath: /var/temp
          name: bar
        - mountPath: /var/tmp
          name: foo
      enableServiceLinks: false
      hostname: container-name
      imagePullSecrets:
      - name: container-name-pull-1234567890ab
      serviceAccountName: container-name
      terminationGracePeriodSeconds: 10
      volumes:
      - name: bar
        persistentVolumeClaim:
          claimName: bar
      - name: foo
        persistentVolumeClaim:
          claimName: foo
status: {}

---
apiVersion: policy/v1
kind: PodDisruptionBudget
metadata:
  annotations:
    acorn.io/config-hash: ""
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/app-public-name: app-name
    acorn.io/container-name: container-name
    acorn.io/managed: "true"
    acorn.io/project-name: app-namespace
  name: container-name
  namespace: app-created-namespace
spec:
  maxUnavailable: 1
  selector:
    matchLabels:
      acorn.io/app-name: app-name
      acorn.io/app-namespace: app-namespace
      acorn.io/container-name: container-name
      acorn.io/managed: "true"
status:
  currentHealthy: 0
  desiredHealthy: 0
  disruptionsAllowed: 0
  expectedPods: 0

---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  annotations:
    acorn.io/config-hash: ""
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/managed: "true"
    acorn.io/public-name: app-name.bar
    acorn.io/volume-class: ""
    acorn.io/volume-name: bar
  name: bar
  namespace: app-created-namespace
spec:
  accessModes:
  - ReadWriteMany
  resources:
    requests:
      storage: 6Gi
  storageClassName: ""
status: {}

---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  annotations:
    acorn.io/config-hash: ""
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/managed: "true"
    acorn.io/public-name: app-name.foo
    acorn.io/volume-class: ""
    acorn.io/volume-name: foo
  name: foo
  namespace: app-created-namespace
spec:
  accessModes:
  - ReadWriteMany
  resources:
    requests:
      storage: 3Gi
  storageClassName: ""
status: {}

---
apiVersion: internal.acorn.io/v1
kind: AppInstance
metadata:
  creationTimestamp: null
  name: app-name
  namespace: app-namespace
  uid: 1234567890abcdef
spec:
  image: test
  volumes:
  - accessModes:
    - readWriteOnce
    class: test-custom-class
    size: 3Gi
    target: foo
  - class: test-custom-class
    target: bar
status:
  appImage:
    buildContext: {}
    id: test
    imageData: {}
    vcs: {}
  appSpec:
    containers:
      container-name:
        dirs:
          /var/temp:
            secret: {}
            volume: bar
          /var/tmp:
            secret: {}
            volume: foo
        image: image-name
        metrics: {}
        probes: null
    volumes:
      bar:
        class: test-custom-class-request
        size: 6Gi
      foo:
        accessModes:
        - readOnlyMany
        class: test-custom-class-request
        size: 6Gi
  appStatus: {}
  columns: {}
  conditions:
    reason: Success
    status: "True"
    success: true
    type: defined
  namespace: app-created-namespace
  resolvedOfferings:
    volumes:
      bar:
        accessModes:
        - readWriteMany
        class: test-custom-class-default
      foo:
        accessModes:
        - readWriteMany
        class: test-custom-class-default
        size: 2Gi
  staged:
    appImage:
      buildContext: {}
      imageData: {}
      vcs: {}
  summary: {}
`
